x-pack/filebeat/docs/inputs/input-cel.asciidoc

```diff
@@ -0,0 +1,500 @@
+[role="xpack"]
+
+:type: cel
+
+[id="{beatname_lc}-input-{type}"]
+=== Common Expression Language input
+
+++++
+<titleabbrev>CEL</titleabbrev>
+++++
+
+Use the `cel` input to read messages from a file path or HTTP API with a variety of payloads using the https://opensource.google.com/projects/cel[Common Expression Language (CEL)] and the https://pkg.go.dev/github.com/elastic/mito/lib[mito] and https://pkg.go.dev/github.com/google/cel-go@v0.12.5/ext#Strings[ext Strings] CEL extension libraries.
+
+CEL is a non-Turing complete language that can perform evaluation of expression in inputs, which can incled with the mito extension library file and API endpoints.
+The `cel` input periodically runs a CEL program that is given an execution environment that may be configured by the user, and publishes the set of events that result from the program evaluation.
+Optionally the CEL program may return cursor states that will be provided to the next execution of the CEL program.
+The cursor states may be used to control the behaviour of the program.
+
+This input supports:
+
+* Auth
+** Basic
+** OAuth2
+* Retrieval at a configurable interval
+* Pagination
+* Retries
+* Rate limiting
+* Proxying
+
+Example configurations:
+
+["source","yaml",subs="attributes"]
+----
+filebeat.inputs:
+# Fetch your public IP every minute.
+- type: cel
+  interval: 1m
+  resource.url: https://api.ipify.org/?format=json
+  program: |
+    bytes(get(state.url).Body).as(body, {
+        "events": [body.decode_json()]
+    })
+----
+
+["source","yaml",subs="attributes"]
+----
+filebeat.inputs:
+- type: cel
+  resource.url: http://localhost:9200/_search
+  state:
+    scroll: 5m
+  program: |
+    (
+        !has(state.cursor) || !has(state.cursor.scroll_id) ?
+            post(state.url+"?scroll=5m", "", "")
+        :
+            post(
+                state.url+"/scroll?"+{"scroll_id": [state.cursor.scroll_id]}.format_query(),
+                "application/json",
+                {"scroll": state.scroll}.encode_json()
+            )
+    ).as(resp, bytes(resp.Body).decode_json().as(body, {
+            "events": body.hits.hits,
+            "cursor": {"scroll_id": body._scroll_id},
+    }))
```

Comment:
I would like to be able to test this with `mito` but am not able to get the connection to work because of the secure connection requirement in conjunction with the "x509: “elasticsearch” certificate is not standards compliant" error.

Using the following inputs
example.json
```
{
	"url": "https://localhost:9200/_search",
	"scroll": "5m"
}
```

example.cel
```
(
    !has(state.cursor) || !has(state.cursor.scroll_id) ?
        post(state.url+"?scroll=5m", "", "")
    :
        post(
            state.url+"/scroll?"+{"scroll_id": [state.cursor.scroll_id]}.format_query(),
            "application/json",
            {"scroll": state.scroll}.encode_json()
        )
).as(resp, bytes(resp.Body).decode_json().as(body, {
        "events": body.hits.hits,
        "cursor": {"scroll_id": body._scroll_id},
}))
```
and
```
$ mito -data example.json example.cel
failed eval: Post "https://localhost:9200/_search?scroll=5m": x509: “elasticsearch” certificate is not standards compliant
```

The correct requests are constructed.

Without cursor state
```
{
	"Body": "",
	"Close": false,
	"ContentLength": 0,
	"Header": {
		"Content-Type": [
			""
		]
	},
	"Host": "localhost:9200",
	"Method": "POST",
	"Proto": "HTTP/1.1",
	"ProtoMajor": 1,
	"ProtoMinor": 1,
	"URL": "https://localhost:9200/_search?scroll=5m"
}
```

With `state.cursor.scroll_id` set to "token":
```
{
	"Body": "{\"scroll\":\"5m\"}",
	"Close": false,
	"ContentLength": 15,
	"Header": {
		"Content-Type": [
			"application/json"
		]
	},
	"Host": "localhost:9200",
	"Method": "POST",
	"Proto": "HTTP/1.1",
	"ProtoMajor": 1,
	"ProtoMinor": 1,
	"URL": "https://localhost:9200/_search/scroll?scroll_id=token"
}
```